symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/JwtHelper.java

```diff
@@ -0,0 +1,105 @@
+package com.symphony.bdk.core.auth;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.apiguardian.api.API;
+import org.bouncycastle.asn1.pkcs.RSAPrivateKey;
+import org.bouncycastle.crypto.params.RSAPrivateCrtKeyParameters;
+import org.bouncycastle.crypto.util.PrivateKeyInfoFactory;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemReader;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+import java.util.Date;
+
+/**
+ * JWT helper class, used to :
+ * <ul>
+ *   <li>load a private key</li>
+ *   <li>generated a signed JWT for a given user</li>
+ * </ul>
+ */
+@API(status = API.Status.INTERNAL)
+public class JwtHelper {
+
+	// PKCS#8 format
+	private static final String PEM_PRIVATE_START = "-----BEGIN PRIVATE KEY-----";
+	private static final String PEM_PRIVATE_END = "-----END PRIVATE KEY-----";
+
+	// PKCS#1 format
+	private static final String PEM_RSA_PRIVATE_START = "-----BEGIN RSA PRIVATE KEY-----";
+
+	/**
+	 * Creates a JWT with the provided user name and expiration date, signed with the provided private key.
+	 * @param user the username to authenticate; will be verified by the pod
+	 * @param expiration of the authentication request in milliseconds; cannot be longer than the value defined on the
+	 * pod
+	 * @param privateKey the private RSA key to be used to sign the authentication request; will be checked on the pod
+	 * against
+	 * the public key stored for the user
+	 * @return a signed JWT for a specific user (or subject)
+	 */
+	public static String createSignedJwt(String user, long expiration, Key privateKey) {
+		return Jwts.builder()
+			.setSubject(user)
+			.setExpiration(new Date(System.currentTimeMillis() + expiration))
+			.signWith(SignatureAlgorithm.RS512, privateKey)
+			.compact();
+	}
+
+	/**
+	 * Creates a RSA Private Key from a PEM String. It supports PKCS#1 and PKCS#8 string formats
+	 * @param pemPrivateKey RSA Private Key content
+	 * @return a {@link PrivateKey} instance
+	 * @throws GeneralSecurityException On invalid Private Key
+	 */
+	public static PrivateKey parseRSAPrivateKey(final String pemPrivateKey) throws GeneralSecurityException {
```

Comment:
It could improve readability to extract two methods here, one for PKCS#8 format, one for // PKCS#1 format